test/integration/scheduler/workload_controller_test.go

```diff
@@ -69,6 +70,134 @@ var _ = ginkgo.Describe("Workload controller with scheduler", func() {
+			util.SetAdmissionCheckActive(ctx, k8sClient, check1, metav1.ConditionTrue)
+
+			check2 = testing.MakeAdmissionCheck("check2").ControllerName("ctrl2").Obj()
+			gomega.Expect(k8sClient.Create(ctx, check2)).Should(gomega.Succeed())
+			util.SetAdmissionCheckActive(ctx, k8sClient, check2, metav1.ConditionTrue)
+
+			check3 = testing.MakeAdmissionCheck("check3").ControllerName("ctrl3").Obj()
+			gomega.Expect(k8sClient.Create(ctx, check3)).Should(gomega.Succeed())
+			util.SetAdmissionCheckActive(ctx, k8sClient, check3, metav1.ConditionTrue)
+
+			clusterQueue = testing.MakeClusterQueue("cluster-queue").
+				AdmissionCheckStrategy(
+					*testing.MakeAdmissionCheckStrategyRule("check1", kueue.ResourceFlavorReference(flavorOnDemand)).Obj(),
+					*testing.MakeAdmissionCheckStrategyRule("check2").Obj(),
+					*testing.MakeAdmissionCheckStrategyRule("check3", kueue.ResourceFlavorReference(reservationFlavor)).Obj()).
+				ResourceGroup(
+					*testing.MakeFlavorQuotas(reservationFlavor).Resource(resourceGPU, "1", "1").Obj(),
+					*testing.MakeFlavorQuotas(flavorOnDemand).Resource(resourceGPU, "5", "5").Obj()).
+				Cohort("cohort").
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, clusterQueue)).To(gomega.Succeed())
+
+			localQueue = testing.MakeLocalQueue("queue", ns.Name).ClusterQueue(clusterQueue.Name).Obj()
+			gomega.Expect(k8sClient.Create(ctx, localQueue)).To(gomega.Succeed())
+		})
+
+		ginkgo.AfterEach(func() {
+			gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+			gomega.Expect(util.DeleteLocalQueue(ctx, k8sClient, localQueue)).To(gomega.Succeed())
+			util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, clusterQueue, true)
+			util.ExpectAdmissionCheckToBeDeleted(ctx, k8sClient, check3, true)
+			util.ExpectAdmissionCheckToBeDeleted(ctx, k8sClient, check2, true)
+			util.ExpectAdmissionCheckToBeDeleted(ctx, k8sClient, check1, true)
+			util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, flavor1, true)
+			util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, flavor2, true)
+		})
+
+		ginkgo.It("the workload should have appropriate AdditionalChecks added", func() {
+			wl := testing.MakeWorkload("wl", ns.Name).
+				Queue("queue").
+				Request(resourceGPU, "3").
+				Obj()
+			wlKey := client.ObjectKeyFromObject(wl)
+
+			ginkgo.By("creating and reserving quota the workload", func() {
+				gomega.Expect(k8sClient.Create(ctx, wl)).To(gomega.Succeed())
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, wlKey, &updatedWl)).Should(gomega.Succeed())
+					g.Expect(workload.HasQuotaReservation(&updatedWl)).Should(gomega.BeTrue(), "should have quota reservation")
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+				gomega.Expect(workload.IsAdmitted(&updatedWl)).To(gomega.BeFalse())
+			})
+
+			ginkgo.By("checking workload's admission checks", func() {
+				gomega.Eventually(func(g gomega.Gomega) {
```

Comment:
Can it be in the same eventually for quota reservation? Although I seem to remember that it's added separately.

Reply:
Fixed